### DISCUSSION TITLE
How to migrate full repository with all branches?

### DISCUSSION THREAD

**Discussion by luisa289803** [community/community#47270] [Repositories] [answered]
### Select Topic Area

Product Feedback

### Body

I need to migrate a github repository with all its branches to a new repository (without deleting the old one) is it possible?

**Comment by Juanescacha** [comment, marked as answer]
Hola veo que hablas español, claro que se puede es super sencillo!

## Hay 2 casos:

### 1) Si el repositorio NO es tuyo, solamente con un FORK ya tienes la copia exacta del repositorio en tu cuenta
- Haces Click en el boton Fork
![image](https://user-images.githubusercontent.com/6394715/218558752-71f40883-0de4-4645-a6ec-aedc76564835.png)
- Y aqui, no seleccionando la caja que dice "copy the main branch only" te aseguras de clonar todas las ramas
![image](https://user-images.githubusercontent.com/6394715/218558788-7aaaba7c-e7fb-48a7-9a16-9dc9a81a46a9.png)

### 2) Si el repositorio es tuyo, lo vamos a realizar todo desde una terminal con git, siguiendo los siguientes pasos:
- Clona el repositorio antiguo en tu computadora local
```git clone --bare https://github.com/usuario/repoantiguo.git```
- Crea un nuevo repositorio en GitHub SIN un README ni un archivo .gitignore
- Dirigete en la terminal a la carpeta del repositorio antiguo clonado
```cd repoantiguo```
- Vincula to clon local con el nuevo repositorio de destino en GitHub:
```git remote add nuevorepo https://github.com/usuario/nuevorepo.git```
- Envia todas las ramas e informacion adicional al nuevo repositorio con solamente dos comandos:
```git push --all nuevorepo```
```git push --tags nuevorepo```

Despues de completar todos estos pasos el nuevo repositorio en GitHub debería tener todas las ramas y etiquetas del repositorio original.

y asi tendrias dos copias de tu mismo repositorio! 

Si tienes cualquier duda con respecto a los pasos @luisa289803 me dices y te explico, espero te pueda servir de ayuda 🙏

**Reply by luisa289803** [reply]
Krack @Juanescacha !!! 
Mil gracias por tu ayuda! efectivamente funcionó la opción 1 ya que los repositorios no son de mi propiedad. 

Hasta ahora me estoy familiarizando a detalle con la herramienta ...

**Reply by Juanescacha** [reply]
es con mucho gusto!! cualquier otra duda me puedes contactar sin molestia!
